docs/ref/models/querysets.txt

```diff
@@ -1951,6 +1951,20 @@ This has a number of caveats though:
   does not retrieve and set the primary key attribute, as ``save()`` does,
   unless the database backend supports it (currently PostgreSQL).
 * It does not work with many-to-many relationships.
+* It always casts ``objs`` to a list thus fully evaluating any generators
+  regardless of ``batch_size``. This is because all objects need to be
+  inspected so that any objects with a manually set primary key are
+  inserted first. If manually set primary keys are not a concern in your use
+  case then you can evaluate only a batch at a time by using
+  ``itertools``::
+
+    from itertools import islice, takewhile, count
+
+    iterable = iter(objs)
+    batches = takewhile(bool,
+                        (list(islice(iterable, batch_size)) for _ in count(0)))
+    for batch in batches:
+        Entry.objects.bulk_create(batch, batch_size)
```

Comment:
What about

```python
iterator = iter(objs)
while True:
    batch = list(itertools.islice(iterator, chunk_size))
    if not batch:
        break
    Entry.objects.bulk_create(batch, batch_size)
```

Reply:
`iter` is not needed if `objs` is a generator.

Reply:
Thanks Simon, I pushed an updated example based on your suggestion.